src/Angular.js

```diff
@@ -1195,7 +1195,12 @@ function toJson(obj, pretty) {
   if (!isNumber(pretty)) {
     pretty = pretty ? 2 : null;
   }
-  return JSON.stringify(obj, toJsonReplacer, pretty);
+  try {
+    return JSON.stringify(obj, toJsonReplacer, pretty);
+  } catch (RangeError) {
+    // In Safari, 'Invalid Date's throw a range error. 'null' is returned in all other browsers.
+    return 'null';
```

Comment:
This means that the whole Object will be serialized as `"null"` even if one sub-property is an invalid date.
This is inconsistent with other browsers and doesn't feel right either.

I think it is more appropriate to handle this inside the `toJsonReplacer()` function.